setup.py

```diff
@@ -1,3 +1,87 @@
-from setuptools import setup
+from setuptools import find_packages, setup
 
-setup()
+extras = {}
+extras["quality"] = [
+    "black ~= 22.10.0",
+    "isort >= 5.5.4",
+    "flake8 >= 3.8.3",
+]
+extras["docs"] = ["sphinx==4.0.0", "sphinx_rtd_theme"]
+extras["test_prod"] = ["pytest", "pytest-xdist", "pytest-subtests", "parameterized"]
+extras["test_dev"] = [
+    "datasets",
+    "evaluate",
+    "transformers",
+    "scipy",
+    "scikit-learn",
+    "deepspeed<0.7.7",
+    "tqdm",
+]
+extras["testing"] = extras["test_prod"] + extras["test_dev"]
+extras["rich"] = ["rich"]
+
+extras["test_trackers"] = ["wandb"]
+extras["dev"] = extras["quality"] + extras["testing"] + extras["rich"]
+
+
+setup(
+    name="trlx",
+    version="v0.4.0-dev",
+    description="Transformer Reinforcement Learning X",
+    long_description=open("README.md", "r", encoding="utf-8").read(),
+    long_description_content_type="text/markdown",
+    keywords="deep learning",
+    license="MIT",
+    author="The CarperAI team",
+    url="https://github.com/carperai/trlx",
+    packages=find_packages("."),
+    entry_points={
+        "console_scripts": [
+            "trlx=trlx.commands.trlx_cli:main",
+        ]
+    },
+    python_requires=">=3.8.0",
+    install_requires=[
+        "numpy>=1.17",
+        "packaging>=20.0",
+        "psutil",
+        "pyyaml",
+        "torch~=1.13.0",
+        "accelerate~=0.15.0",
+        "typing-extensions~=3.10.0",
+        "torchtyping~=0.1.4",
+        "ray"
```

Comment:
Do we need to duplicate the requirements that already exist in the `setup.cfg` metadata?